snooty/parser.py

```diff
@@ -1179,6 +1180,27 @@ def __init__(
 
         self.config.substitution_nodes = substitution_nodes
 
+        # Parse banner value and instantiate a banner node for postprocessing, if a banner value is defined.
+        for banner in self.config.banners:
+            if banner.value:
+
+                options = {"variant": banner.variant}
+                banner_node = BannerInstantiated(
+                    banner.targets,
+                    n.Directive((-1,), [], "mongodb", "banner", [], options),
+                )
+
+                page, banner_diagnostics = parse_rst(inline_parser, root, banner.value)
+                banner_node.node.children = [
+                    util.fast_deep_copy(child) for child in page.ast.children
```

Comment:
Do we need to duplicate the nodes here? It would be better to duplicate them when inserting them into the page